### ISSUE TITLE
submariner-operator ServiceAccount requires permissions to list DaemonSets

### ISSUE THREAD

**Issue by nwebb11** [submariner-io/submariner-operator#2450]
**What happened**: Using 0.14.1, ran `subctl join  ~/.kube/multi-k8s/cluster-2 broker-info.subm --clusterid multi-cluster-2` to join the cluster, but after deploying the Submarnier operator nothing else was deployed and the logs were showing: 

```
ERR .._networkdiscovery.go:40 ..troller_submariner Error trying to discover network error="error listing the Daemonsets: daemonsets.apps is forbidden: User \"system:serviceaccount:submariner-operator:submariner-operator\" cannot list resource \"daemonsets\" in API group \"apps\" in the namespace \"kube-system\""
```

**What you expected to happen**: Submariner operator to deploy the resources for Submariner.

**How to reproduce it (as minimally and precisely as possible)**: Run `subctl join` to join a cluster, using Submariner, Operator & Subctl with version 0.14.1.

**Anything else we need to know?**: Cilium 1.12.5 was used as the CNI - I'm aware it's not supported. The fix - add the following to the existing ClusterRole `submariner-operator`:

```
- apiGroups:
  - apps
  resources:
  - daemonsets
  verbs:
  - list
```

**Environment**:
- Diagnose information (use `subctl diagnose all`): Not applicable, as Submariner did not deploy
- Gather information (use `subctl gather`):

```
Cluster "multi-cluster-2"
Gathering information from cluster "multi-cluster-2"
 ✓ Gathering operator logs
 ✓ Found 1 pods matching label selector "name=submariner-operator"
 ✓ Gathering operator resources
 ✓ Found 1 submariners in namespace "submariner-operator"
 ✓ Found 0 servicediscoveries in namespace "submariner-operator"
 ✓ Found 1 deployments by field selector "metadata.name=submariner-operator" in namespace "submariner-operator"
 ✓ Found 0 daemonsets by label selector "app=submariner-gateway" in namespace "submariner-operator"
 ✓ Found 0 daemonsets by label selector "app=submariner-routeagent" in namespace "submariner-operator"
 ✓ Found 0 daemonsets by label selector "app=submariner-globalnet" in namespace "submariner-operator"
 ✓ Found 0 deployments by label selector "app=submariner-networkplugin-syncer" in namespace "submariner-operator"
 ✓ Found 0 deployments by label selector "app=submariner-lighthouse-agent" in namespace "submariner-operator"
 ✓ Found 0 deployments by label selector "app=submariner-lighthouse-coredns" in namespace "submariner-operator"
 ✓ Gathering connectivity logs
 ✓ Found 0 pods matching label selector "app=submariner-gateway"
 ✓ Found 0 pods matching label selector "app=submariner-routeagent"
 ✓ Found 0 pods matching label selector "app=submariner-globalnet"
 ✓ Found 0 pods matching label selector "app=submariner-networkplugin-syncer"
 ✓ Found 0 pods matching label selector "app=submariner-addon"
 ✓ Gathering connectivity resources
 ✓ Gathering CNI data from 0 pods matching label selector "app=submariner-routeagent"
 ✓ Gathering CNI data from 0 pods matching label selector "app=submariner-gateway"
 ✓ Gathering globalnet data from 0 pods matching label selector "app=submariner-globalnet"
 ✓ Gathering cable driver data from 0 pods matching label selector "app=submariner-gateway"
 ✓ Found 0 endpoints in namespace "submariner-operator"
 ✓ Found 0 clusters in namespace "submariner-operator"
 ✓ Found 0 gateways in namespace "submariner-operator"
 ✓ Found 0 clusterglobalegressips in namespace ""
 ✓ Found 0 globalegressips in namespace ""
 ✓ Found 0 globalingressips in namespace ""
 ⚠ Gathering service-discovery logs
 ⚠ The Submariner service discovery components are not installed
 ⚠ Gathering service-discovery resources
 ⚠ The Submariner service discovery components are not installed
 ✓ Gathering broker logs
 ✓ Gathering broker resources
 ✓ Found 0 endpoints in namespace "submariner-k8s-broker"
 ✓ Found 0 clusters in namespace "submariner-k8s-broker"
 ✓ Found 0 endpointslices by label selector "endpointslice.kubernetes.io/managed-by=lighthouse-agent.submariner.io" in namespace "submariner-k8s-broker"
 ✓ Found 0 serviceimports in namespace "submariner-k8s-broker"
Files are stored under directory "submariner-20230109140900/multi-cluster-2"
```

- Cloud provider or hardware configuration: Kubeadm & Kubernetes 1.25.5
- Install tools: Subctl
- Others: Cilium 1.12.5

**Comment by dfarrell07** [comment]
I wonder if the code branch that Cilium triggers uses the wrong client. @tpantelis volunteered to take a look. Thanks for the report!

**Comment by tpantelis** [comment]
The error emanates from the `flannel` CNI discovery which searches for the flannel `DaemonSet` in the _kube-system_ namespace. This was added by https://github.com/submariner-io/submariner-operator/pull/1850 which was merged relatively late in the 0.14 release cycle. `Flannel` discovery is tried as the next-to-last resort before the generic discovery and this path isn't tested by CI so it slipped thru the cracks.

**Comment by liuxuzxx** [comment]
Cilium CNI also has the error: error discovering cluster network: error listing the Daemonsets: daemonsets.apps is forbidden
